File: ReadingTracker/app/src/main/java/com/codelab/readingtracker/Entry.java
package com.codelab.readingtracker;

import androidx.annotation.NonNull;

public class Entry {
    public String title;
    public int pageNumber;

    public Entry(String title, int pageNumber) {
        this.title = title;
        this.pageNumber = pageNumber;
    }

    @NonNull
    @Override
    public String toString() {
        return title + "," + pageNumber;
    }

    public static Entry fromString(String entry) {
        String[] fields = entry.split(",");
        String title = fields[0];
        int pageNumber = Integer.parseInt(fields[1]);
        return new Entry(title, pageNumber);
    }
}

File: ReadingTracker/app/src/main/java/com/codelab/readingtracker/reading/Entry.kt
package com.codelab.readingtracker.reading

class Entry(var title: String, var pageNumber: Int) {

    override fun toString(): String {
        return "$title,$pageNumber"
    }

    companion object {

        fun fromString(entry: String): Entry {
            val fields = entry.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            val title = fields[0]
            val pageNumber = Integer.parseInt(fields[1])
            return Entry(title, pageNumber)
        }
    }
}
